csp_billing_adapter_local/plugin.py

```diff
@@ -13,66 +13,102 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+"""
+plugin.py is part of csp-billing-adapter-local and provides the local storage
+plugin
+"""
 
 import json
-import os
+import logging
+
+from json.decoder import JSONDecodeError
+from pathlib import Path
 
 import csp_billing_adapter
 
 from csp_billing_adapter.config import Config
 
-cache_file = os.path.expanduser('~/csp-adapter/cache.json')
-csp_config_file = os.path.expanduser('~/csp-adapter/csp-config.json')
+log = logging.getLogger('CSPBillingAdapter')
+
+LOCAL_DIR = '.local'
+PRODUCT_DIR = 'csp-billing-adapter'
+CACHE_FILE = 'cache.json'
+CSP_CONFIG_FILE = 'csp-config.json'
```

Comment:
Hmm, do we have another config file that only configures behavior specific to the csp?

Reply:
I am not sure I follow. This was how @smarlowucf originally had it setup. Perhaps he can comment. The file contains

```
{"billing_api_access_ok": true, "timestamp": "2023-04-21T20:14:25.188927+00:00", "expire": "2023-04-21T21:14:25.188927+00:00", "errors": []}
```
maybe it just needs a new name?
which is different

Reply:
That looks similar to what was discussed w.r.t. the data the adapter would make available to the application for consumption, primarily intended for supportconfig generation. As such I think the `-config` is misleading as this is not a configuration file.

Reply:
csp-billing-state.json?  csp-adapter-state.json?  Please let me know yours and @smarlowucf  preference .

Reply:
`csp-adapter-state.json` is more descriptive to what we are communicating, for me but either works

Reply:
@rjschwei I can change the file name, but the actual hooks are called get_csp_config/update_csp_config/save_csp_config and to change that means in all plugins. any thoughts @smarlowucf @rtamalin 

https://github.com/SUSE-Enceladus/csp-billing-adapter/blob/devel/csp_billing_adapter/csp_config.py
https://github.com/SUSE-Enceladus/csp-billing-adapter/blob/devel/csp_billing_adapter/storage_hookspecs.py

Reply:
csp-config has been the name for the configMap since we started the project. It was pulled over from the Rancher solution and has been in every design doc related to NV. There's been plenty of time to change it up to now. However, this is a public interface and changing it now will affect the adapter code and NV code. Thus my suggestion is we stick to it as is. If we don't like the name it changes in v2. We don't have the time for changing a public interface at this point.

Reply:
@smarlowucf @rtamalin any preferences/comments on  changing the dir from ~/.local/csp-billing-adapter to /var/lib/csp-billing-adapter?

Reply:
Nope